code/modules/integrated_electronics/core/assemblies.dm

```diff
@@ -40,11 +41,12 @@
 
 /obj/item/device/electronic_assembly/proc/handle_idle_power()
 	// First we generate power.
-	for(var/obj/item/integrated_circuit/passive/power/P in contents)
+	for(var/obj/item/integrated_circuit/passive/power/P in components)
```

Comment:
Could you name this circuit_components or assembly_components ? So it's more obvious these are not generic ones.